# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/ffmpeg/FFmpegDataDecoder.h
# Commit: e9ba11d2516b
# Full Hash: e9ba11d2516b9df4de51f07aba224c42850d1b8a
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2020-03-02 21:27:32
# Regressor Bug: 1616185
# File Overlap Count: 1
# Description:
#   Bug 1616185 [Wayland] Implement VA-API decode in FFmpegDataDecoder, r=jya
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D63132
# ==============================================================================

diff -r 097aecdda3d0 -r e9ba11d2516b dom/media/platforms/ffmpeg/FFmpegDataDecoder.h
--- a/dom/media/platforms/ffmpeg/FFmpegDataDecoder.h	Mon Mar 02 10:48:57 2020 +0000
+++ b/dom/media/platforms/ffmpeg/FFmpegDataDecoder.h	Mon Mar 02 11:41:05 2020 +0000
@@ -47,6 +47,7 @@
   virtual void InitCodecContext() {}
   AVFrame* PrepareFrame();
   MediaResult InitDecoder();
+  MediaResult AllocateExtraData();
   MediaResult DoDecode(MediaRawData* aSample, bool* aGotFrame,
                        DecodedData& aOutResults);
 
@@ -67,10 +68,10 @@
   virtual bool NeedParser() const { return false; }
   virtual int ParserFlags() const { return PARSER_FLAG_COMPLETE_FRAMES; }
 
-  static StaticMutex sMonitor;
   const RefPtr<TaskQueue> mTaskQueue;
   MozPromiseHolder<DecodePromise> mPromise;
   media::TimeUnit mLastInputDts;
+  static StaticMutex sMonitor;
 };
 
 }  // namespace mozilla
